src/IssueViz.UnitTests/Editor/SelectedIssueTagging/Adornment/IssueLocationAdornmentTagger_CacheTests.cs

```diff
@@ -0,0 +1,177 @@
+            var lineSource2 = CreateFormattedLineSource(18d);
+            var textView = CreateWpfTextView(ValidSnapshot, lineSource1);
+            var testSubject = new CachingAdornmentFactory(textView);
+
+            var locViz = CreateLocationViz();
+
+            var actual1 = testSubject.CreateOrUpdate(locViz);
+            actual1.FontSize.Should().Be(12d); // sanity check
+
+            ChangedMockedLineSource(textView, lineSource2);
+
+            // Act
+            var actual2 = testSubject.CreateOrUpdate(locViz);
+            actual2.FontSize.Should().Be(18d);
+        }
+
+        [TestMethod]
+        public void RemoveUnused_EmptyCache_NoError()
+        {
+            var testSubject = new CachingAdornmentFactory(ValidTextView);
+
+            Action act = () => testSubject.RemoveUnused(new[] { ValidLocViz });
+            act.Should().NotThrow();
+        }
+
+        [TestMethod]
+        public void RemoveUnused_NoCurrentLocations_AllRemoved()
+        {
+            var testSubject = new CachingAdornmentFactory(ValidTextView);
+            testSubject.CreateOrUpdate(CreateLocationViz());
+            testSubject.CreateOrUpdate(CreateLocationViz());
+            testSubject.CreateOrUpdate(CreateLocationViz());
+            testSubject.CachedAdornments.Count.Should().Be(3); // sanity check
+
+            testSubject.RemoveUnused(Array.Empty<IAnalysisIssueLocationVisualization>());
+            testSubject.CachedAdornments.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void RemoveUnused_AllLocationsAreCurrent_NoAdormentsRemoved()
+        {
+            var testSubject = new CachingAdornmentFactory(ValidTextView);
+
+            var locViz1 = CreateLocationViz();
+            var locViz2 = CreateLocationViz();
+
+            var adornment1 = testSubject.CreateOrUpdate(locViz1);
+            var adornment2 = testSubject.CreateOrUpdate(locViz2);
+            var expectedAdornments = new[] { adornment1, adornment2 };
+
+            // Act
+            testSubject.RemoveUnused(new[] { locViz1, locViz2 });
+            testSubject.CachedAdornments.Should().BeEquivalentTo(expectedAdornments);
+        }
+
+        [TestMethod]
+        public void RemoveUnused_NotAllLocationsAreCurrent_ExpectedEntriesRemoved()
+        {
+            var testSubject = new CachingAdornmentFactory(ValidTextView);
+
+            var locViz1 = CreateLocationViz();
+            var locViz2 = CreateLocationViz();
+            var locViz3 = CreateLocationViz();
+
+            var adornment1 = testSubject.CreateOrUpdate(locViz1);
+            var adornment2 = testSubject.CreateOrUpdate(locViz2);
+            var adornment3 = testSubject.CreateOrUpdate(locViz3);
+            testSubject.CachedAdornments.Should().BeEquivalentTo(new[] { adornment1, adornment2, adornment3 }); // sanity check
+
+            // Remove multiple items
+            testSubject.RemoveUnused(new[] { locViz3 });
```

Comment:
Btw, the usage really implies that the unused is `locViz3`.

Reply:
Calling it `Refresh` helps, but it's still odd to pass in the list of things you want to keep.

Reply:
Hm, maybe `Set` then?

Reply:
Seems odd to call `Initialize` multiple times, and `Set` and `Initialize` both imply that the cache is being populated i.e. new items created if necessary, which isn't the case. I think `Refresh` is good enough.